src/android.ts

```diff
@@ -295,6 +296,22 @@ export const androidApi = <RunTarget extends SupportedRunTarget<'android'>>(
     },
     async ensureDevice() {
         await this._internal.ensureAdb();
+
+        const availableDevices = await listDevices({ frida: options.capabilities.includes('frida') });
+
+        if (availableDevices.length > 1)
+            throw new Error('You have multiple devices connected. Please disconnect all but one.');
+        else if (availableDevices.length === 0)
+            throw new Error(
+                options.runTarget === 'device' ? 'You need to connect your device.' : 'You need to start the emulator.'
+            );
+        else if (availableDevices.filter((device) => device.platform === 'ios').length === 0)
+            throw new Error(
+                options.runTarget === 'device'
+                    ? 'You need to connect an Android device.'
+                    : 'You need to start the emulator.'
+            );
+
         if (
```

Comment:
Shouldn't we have a different error (that mentions that the device needs to be booted) here now?